tests/scalers/redis/redis_cluster_streams_lag/redis_cluster_streams_lag_test.go

```diff
@@ -0,0 +1,235 @@
+        command: ["./main"]
+        args: ["consumer"]
+        env:
+        - name: REDIS_MODE
+          value: CLUSTER
+        - name: REDIS_HOSTS
+          value: {{.RedisHost}}.{{.RedisNamespace}}
+        - name: REDIS_PORTS
+          value: "6379"
+        - name: REDIS_STREAM_NAME
+          value: my-stream
+        - name: REDIS_STREAM_CONSUMER_GROUP_NAME
+          value: consumer-group-1
+        - name: REDIS_PASSWORD
+          value: {{.RedisPassword}}
+`
+
+	secretTemplate = `apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+type: Opaque
+data:
+  password: {{.RedisPasswordBase64}}
+`
+
+	triggerAuthenticationTemplate = `apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthenticationName}}
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+  - parameter: password
+    name: {{.SecretName}}
+    key: password
+`
+
+	scaledObjectTemplate = `apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  pollingInterval: 5
+  cooldownPeriod:  10
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  advanced:
+    horizontalPodAutoscalerConfig:
+      behavior:
+        scaleDown:
+          stabilizationWindowSeconds: 15
+  triggers:
+  - type: redis-cluster-streams
+    metadata:
+      hostsFromEnv: REDIS_HOSTS
+      portsFromEnv: REDIS_PORTS
+      stream: my-stream
+      consumerGroup: consumer-group-1
+      pendingEntriesCount: "10"
+      lagCount: "4"
+    authenticationRef:
+      name: {{.TriggerAuthenticationName}}
+`
+
+	insertJobTemplate = `apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{.JobName}}
+  namespace: {{.TestNamespace}}
+spec:
+  ttlSecondsAfterFinished: 0
+  template:
+    spec:
+      containers:
+      - name: redis
+        image: ghcr.io/kedacore/tests-redis-streams:latest
+        imagePullPolicy: IfNotPresent
+        command: ["./main"]
+        args: ["producer"]
+        env:
+        - name: REDIS_MODE
+          value: CLUSTER
+        - name: REDIS_HOSTS
+          value: {{.RedisHost}}.{{.RedisNamespace}}
+        - name: REDIS_PORTS
+          value: "6379"
+        - name: REDIS_STREAM_NAME
+          value: my-stream
+        - name: REDIS_PASSWORD
+          value: {{.RedisPassword}}
+        - name: NUM_MESSAGES
+          value: "{{.ItemsToWrite}}"
+      restartPolicy: Never
+  backoffLimit: 4
+`
+)
+
+func TestScaler(t *testing.T) {
+	// Create kubernetes resources for PostgreSQL server
+	kc := GetKubernetesClient(t)
+
+	// Create Redis Cluster
+	redis.InstallCluster(t, kc, testName, redisNamespace, redisPassword)
+
+	// Create kubernetes resources for testing
+	data, templates := getTemplateData()
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	testScaleOut(t, kc, data)
+	testScaleIn(t, kc)
```

Comment:
As this scaler will allow scaling to zero, we should check the activation phase as part of the e2e tests whenever the scalign to zero is allowed (in this case it is, in the old case it isn't). The activation test is basically setting an `activation*` bigger than 0 (like 5) and queuing 4 elements. After 1 minute without scaling from zero we can guarantee that the activation value is working. The rest of the test is like right now, just queuing some extra messages for scaling out/in
You can check redis list as example: https://github.com/kedacore/keda/blob/main/tests/scalers/redis/redis_standalone_lists/redis_standalone_lists_test.go#L185-L191